Utility/PiecesTemplates.FCMacro

```diff
@@ -0,0 +1,421 @@
+
+        if (self.longitudListaGrupos == 0):
+            self.form.comboPosicionGrupo.addItem(tr("el primero"))
+            self.form.eliminarGrupo.setEnabled(False)
+        else:
+            max = self.longitudListaGrupos
+            for cont in range(max):
+                if cont == 0:
+                    self.form.comboPosicionGrupo.addItem(tr("el primero"))
+                if (cont !=0) and (cont < max):
+                    self.form.comboPosicionGrupo.addItem(str(cont+1))
+                if (cont == max-1):
+                    self.form.comboPosicionGrupo.addItem(tr("el último"))                    
+
+    def nombrePiezaCambia(self):
+        if (self.form.edicionNombrePieza.text() != ''):
+            self.form.botonCrear.setEnabled(True)
+        else:
+            self.form.botonCrear.setEnabled(False)
+
+    def on_eliminarGrupo_clicked(self):
+        # Delete Group
+        aBorrar = self.form.comboEliminarGrupo.currentText()
+        listGroups.remove(aBorrar)
+        elementosBorrados.append(aBorrar)       
+        self.longitudListaGrupos = len(listGroups)        
+        self.mostrarCheckbox()
+        self.refreshFichero()
+
+    def on_anadirGrupo_clicked(self):
+        # Add Group  
+        nombrePos = self.form.comboPosicionGrupo.currentText()
+        if (nombrePos == "el primero"):
+            posicion = 1
+        elif (nombrePos == "el último"):
+            posicion = self.longitudListaGrupos+1
+        else:
+            posicion = int(self.form.comboPosicionGrupo.currentText())
+        if self.longitudListaGrupos == 0:
+            listGroups.append(self.form.leAnadirGrupo.text()) 
+        else:
+            listGroups.insert(posicion-1,self.form.leAnadirGrupo.text())
+        self.longitudListaGrupos = len(listGroups)
+        
+        self.mostrarCheckbox()
+        self.refreshFichero()
+        self.form.eliminarGrupo.setEnabled(True)
+
+    def refreshFichero(self):
+        # Refresh File (destroy and create the same file)
+        groupsFile = open(gr_path, "w")
+        for cont in range(self.longitudListaGrupos):
+            groupsFile.write(listGroups[cont]+"\n")
+        groupsFile.close()
+
+        self.posGroup()
+        self.mostrarCheckbox()
+        self.mostrarComboGroups()
+        self.form.leAnadirGrupo.setText("")           
+
+    def createPart(self): 
+        # Create one piece with groups   
+        self.nombrePieza = self.form.edicionNombrePieza.text()
+        App.activeDocument().Tip = App.activeDocument().addObject('App::Part','Part')
+        App.ActiveDocument.recompute()
+
+        listaChecked = []
+        if self.longitudListaGrupos < maxElem:
+            longChecked = maxElem
+        else:
+            longChecked = self.longitudListaGrupos
+        for cont in range(longChecked):
+            listaChecked.append(True)
+
+        for nombrePart in FreeCAD.ActiveDocument.Objects:
+            if nombrePart.Name.find('Part') > -1:
```

Comment:
This is very fragile, it'll break if the user has an object with `Part` in its name. Use a global variable if necessary to save the object (not its name but really the object).
Use at least `Part in nombrePart.Name`.